assemble/conf/create-jshell.sh

```diff
@@ -0,0 +1,134 @@
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+function addLicenseHeader(){
+  export header="$1"
+  export jPath="$2"
+  
+  # Does an auto-generated JShell config file exists?
+  if [[ -e "$jPath" ]]; then
+    printf "%s\n" "/*" >> "$jPath"
+    
+    # Load in license header
+    while IFS= read -r line; do
+      echo "$line" >> "$jPath"
+    done < "$header"
+    
+    printf "%s\n" "*/" >> "$jPath" 
+    echo " " >> "$jPath"
+  else
+    echo "Cannot add license header in $jPath"
+    echo "Please ensure jshell-init.jsh exists"
+    exit 1
+  fi
+}
+
+function addAccumuloAPI(){
+  export srcDir="$1"
+  export jPath="$2"
+  export client="accumulo/core/client"
+  export data="accumulo/core/data"
+  export security="accumulo/core/security"
+  export mini="accumulo/minicluster"
+  export hadoop="accumulo/hadoop"
+    
+  # Does an auto-generated JShell config file exists?
+  if [[ ! -e "$jPath" ]]; then
+    echo "Cannot add APIs in $jPath"
+    echo "Please ensure jshell-init.jsh exists"
+    exit 1
+  fi
+  
+  # Is the source directory valid?
+  if [[ ! -d "$srcDir" ]]; then
+    echo "$srcDir is not a valid directory. Please make sure it exists."
+    rm "$jPath"
+    exit 1
+  fi
+  
+  # Does a valid JShell path and source directory exists?
+  if [[ -e "$jPath" ]] && [[ -d "$srcDir" ]]; then
+    # Add API category designator in jshell-init.jsh 
+    case "$srcDir" in
+        *"$client"*) echo "// Accumulo Client API" >> "$jPath";;
+        *"$data"*) echo "// Accumulo Data API" >> "$jPath";;
+        *"$security"*) echo "// Accumulo Security API" >> "$jPath";;
+        *"$mini"*) echo "// Accumulo Minicluster API" >> "$jPath";;
+        *"$hadoop"*) echo "// Accumulo Hadoop API" >> "$jPath";;
+        *) echo "// Other API" >> "$jPath";;
+    esac
+   
+    # Extract API info from provided source directory
+    mapfile -t api < <(find "$srcDir" -type f -name '*.java'| 
+                       xargs -n1 dirname| sort -u)
+   
+    # Load in API and format source directory into Java import statements
+    for apiPath in "${api[@]}"; do
+       printf "%s\n" "import ${apiPath##*/java/}.*" >> "$jPath"
+    done
+    sed -i '/^ *import / s#/#.#g' "$jPath"
+    echo " " >> "$jPath"
+  fi
+}
+
+function main(){
+  # Establish Accumulo's main base directory
+  SOURCE="${BASH_SOURCE[0]}"
+  while [[ -h "${SOURCE}" ]]; do
+    bin="$( cd -P "$( dirname "${SOURCE}" )" && pwd )"
+    SOURCE="$(readlink "${SOURCE}")"
+    [[ "${SOURCE}" != /* ]] && SOURCE="${bin}/${SOURCE}"
+  done
+    
+  # Establish file and folder paths for JShell config
+  export conf="$( cd -P "$( dirname "${SOURCE}" )" && pwd )"; 
+  export jPath="$conf/jshell-init.jsh"
```

Comment:
You should generate this in the `assemble/target` directory during the build, and then update the `assemble/src/main/assemblies/component.xml` file to ensure it is copied to the appropriate `conf/` directory by the `maven-assembly-plugin` for the distribution tarball.